src/ts/lib/util/common.ts

```diff
@@ -945,6 +951,79 @@ class UtilCommon {
 		return String(s || '').replace(/&/g, '&amp;').replace(/</g, '&lt;').replace(/>/g, '&gt;');
 	};
 
+	private getPrismComponentDependencies (lang: string) {
+		const components = require(`prismjs/components.js`);
```

Comment:
You are doing the same require 3 times in 3 different methods. Also there are 2 methods just to get a list of components, it's overcomplicated.